client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -580,6 +581,36 @@ public void testFlush() {
         assertThat(request.getMethod(), equalTo(HttpPost.METHOD_NAME));
     }
 
+    public void testForceMerge() {
+        String[] indices = randomIndicesNames(0, 5);
+        ForceMergeRequest forceMergeRequest = new ForceMergeRequest(indices);
+        Map<String, String> expectedParams = new HashMap<>();
+        setRandomIndicesOptions(forceMergeRequest::indicesOptions, forceMergeRequest::indicesOptions, expectedParams);
+        if (randomBoolean()) {
+            forceMergeRequest.maxNumSegments(randomInt());
+        }
+        expectedParams.put("max_num_segments", Integer.toString(forceMergeRequest.maxNumSegments()));
+        if (randomBoolean()) {
+            forceMergeRequest.onlyExpungeDeletes(randomBoolean());
+        }
+        expectedParams.put("only_expunge_deletes", Boolean.toString(forceMergeRequest.onlyExpungeDeletes()));
+        if (randomBoolean()) {
+            forceMergeRequest.flush(randomBoolean());
+        }
+        expectedParams.put("flush", Boolean.toString(forceMergeRequest.flush()));
+
+        Request request = Request.forceMerge(forceMergeRequest);
+        StringJoiner endpoint = new StringJoiner("/", "/", "");
+        if (indices.length > 0) {
+            endpoint.add(String.join(",", indices));
+        }
+        endpoint.add("_forcemerge");
+        assertThat(request.getEndpoint(), equalTo(endpoint.toString()));
+        assertThat(request.getParameters(), equalTo(expectedParams));
+        assertThat(request.getEntity(), nullValue());
+        assertThat(request.getMethod(), equalTo(HttpPost.METHOD_NAME));
```

Comment:
can you add a test for the case when indices are not set or set to null? I found also in the clear cache PR that things break :) (also with refresh and flush)

Reply:
Hi @javanna, when indices are not set or set to `null`, we need to do operations on all the indices ? or throw an exception ? According to our [docs](https://github.com/elastic/elasticsearch/pull/28896/files#diff-442b78b5f72882e641618817047ffb91R785) it should be the former ? But I see currently if we didn't set indices (null), we will get an exception when we build the endpoint [here](https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java#L594), shall we need to modify it ?

Reply:
we should use the endpoint that requires no indices (e.g. `/_forcemerge`), the server will expand that to all indices.